tools/convert_to_16p2

```diff
@@ -66,19 +63,27 @@ ie. 16-bit yuv planar data samples with the active bits in the high order bits
 
     args = parser.parse_args()
 
-    f = open(args.infilename, "rb")
-
     outfilename = args.infilename + ".16p2"
+    of = open(outfilename, "wb")
 
-    data = bytearray(f.read())
+    word = bytearray(2)
+    temp = bytearray(1)
+    with open(args.infilename, "rb") as fi:
+        with open(args.infilename+'.16p2', "wb") as fo:
+            while True:
+                if fi.readinto(word) == 0:
+                    break
+                if args.bits == 8:
+                    temp = word[1]
+                    word[1] = 0xFF
 
-    if args.bits == 8:
-        data = struct.unpack("<" + ("B" * len(data)), data)
-    else:
-        data = struct.unpack("<" + ("H" * (len(data)//2)), data)
+                    fo.write(word)
 
-    data = [ x << (16 - args.bits) for x in data ]
-    
-    data = b"".join(( b"%c%c" % ((x >> 8)&0xFF, x&0xFF) for x in data))
-    of = open(outfilename, "wb")
-    of.write(data)
+                    word[0] = temp
+                    fo.write(word)
```

Comment:
The above sequence could probably be made clearer by writing it as something like:

```python
fo.write(word[0:1])
fo.write(b"\x00")
fo.write(word[1:2])
fo.write(b"\x00")
```

(Though do check this a: doesn't tank performance and b: actually does work(!))